### ISSUE TITLE
Modifying values() tensor of COO tensor requiring grad throws an odd error message

### ISSUE THREAD

**Issue by mruberry** [pytorch/pytorch#48227]
```
a = torch.randn((2, 2)).to_sparse()
a.requires_grad_(True)
a.values()[0] = 0
: RuntimeError: sparse tensors do not have strides
```

While the error's statement is true it's not related to the issue. For example, this works fine when the tensor doesn't require grad:

```
a = torch.randn((2, 2)).to_sparse()
a.values()[0] = 0
a
: tensor(indices=tensor([[0, 0, 1, 1], [0, 1, 0, 1]]),
       values=tensor([ 0.0000,  0.8869, -0.3072, -0.9301]),
       size=(2, 2), nnz=4, layout=torch.sparse_coo)
```

Note that the error is kind of consistent with strided tensors:

```
a = torch.randn((2, 2), requires_grad=True)
a[0][0] = 0
a.sum().backward()
: RuntimeError: leaf variable has been moved into the graph interior
```

Although the error message in this case is, at best, hard to parse for non-experts. Note also that this error message appears when attempting to backward, not when making the assignment like in the sparse snippet.

**Comment by albanD** [comment]
> Note that the error is kind of consistent with strided tensors:

The error "leaf variable has been moved into the graph interior" has been changed in master and will now read "a view of a leaf variable is being modified by an inplace operation" (or something similar) which will be more helpful.


For the original issue, once we decide if the sparse Tensor should be elementary or not, we can start fixing the view/inplace behavior for it (here I expect that we only need to change a flag to make it a non-strided view).

**Comment by SsnL** [comment]
The error is thrown from the graph rewriting code for inplace op on views IIRC, which is before the "leaf variable has been moved into the graph interior" check. The error message should be fixed and maybe recommending operating a copy of values and reconstruct the sparse tensor.
